tools/Dashboard/DevHome.Dashboard/Extensions/ServiceExtensions.cs

```diff
@@ -1,18 +1,23 @@
 ﻿// Copyright (c) Microsoft Corporation and Contributors
 // Licensed under the MIT license.
 
+using DevHome.Dashboard.Services;
 using DevHome.Dashboard.ViewModels;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
 
 namespace DevHome.Settings.Extensions;
+
 public static class ServiceExtensions
 {
     public static IServiceCollection AddDashboard(this IServiceCollection services, HostBuilderContext context)
     {
         // View-models
         services.AddSingleton<DashboardViewModel>();
 
+        // Services
+        services.AddSingleton<WidgetHostingService>();

```

Comment:
In order for the classes that depend on `WidgetHostingService` to be testable, the latter should be mockable. To do that, I suggest registering `IWidgetHostingService` to map to `WidgetHostingService`.

Reply:
#1724